packages/experimental/src/smart-session/utils/CosignerService.ts

```diff
@@ -38,26 +37,67 @@ export async function sendCoSignerRequest<
   transformRequest?: (data: TRequest) => unknown
 }): Promise<TResponse> {
   try {
+    // Transform request data if a transform function is provided
     const transformedData = transformRequest ? transformRequest(request) : request
-    const response = await axios.post<TResponse>(url, transformedData, {
-      params: queryParams,
-      headers
-    })
 
-    return response.data
+    // Construct query string
+    const queryString = new URLSearchParams({
+      ...queryParams
+    }).toString()
+
+    const fullUrl = queryString ? `${url}?${queryString}` : url
+
+    // Prepare fetch options
+    const fetchOptions: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        ...headers
+      },
+      body: JSON.stringify(transformedData)
+    }
+
+    // Perform the fetch request
+    const response = await fetch(fullUrl, fetchOptions)
+
+    // Check if the response is ok (status in the range 200-299)
+    if (!response.ok) {
+      // Try to parse error response
+      const errorData = await response.text()
+      throw new CoSignerApiError(response.status, errorData)
+    }
+
+    // Handle void responses explicitly
+    const contentLength = response.headers.get('content-length')
```

Comment:
Should we handle `null` or `undefined` cases for this to ensure they have valid values?

```js
const contentLength = response.headers.get('content-length') || '0';
const contentType = response.headers.get('content-type') || '';
```

Reply:
The method `isResponseEmptyOrInvalid` will handle this, so might not need to add valid values.